src/load-generator/locustfile.py

```diff
@@ -107,76 +120,112 @@ def get_flagd_value(FlagName):
+            logging.info(f"User getting recommendations for product: {product}")
+            params = {
+                "productIds": [product],
+            }
+            self.client.get("/api/recommendations", params=params)
 
     @task(3)
     def get_ads(self):
-        params = {
-            "contextKeys": [random.choice(categories)],
-        }
-        self.client.get("/api/data/", params=params)
+        category = random.choice(categories)
+        with self.tracer.start_as_current_span("user_get_ads", attributes={"category": str(category)}):
+            logging.info(f"User getting ads for category: {category}")
+            params = {
+                "contextKeys": [category],
+            }
+            self.client.get("/api/data/", params=params)
 
     @task(3)
     def view_cart(self):
-        self.client.get("/api/cart")
+        with self.tracer.start_as_current_span("user_view_cart"):
+            logging.info("User viewing cart")
+            self.client.get("/api/cart")
 
     @task(2)
     def add_to_cart(self, user=""):
         if user == "":
             user = str(uuid.uuid1())
         product = random.choice(products)
-        self.client.get("/api/products/" + product)
-        cart_item = {
-            "item": {
-                "productId": product,
-                "quantity": random.choice([1, 2, 3, 4, 5, 10]),
-            },
-            "userId": user,
-        }
-        self.client.post("/api/cart", json=cart_item)
+        quantity = random.choice([1, 2, 3, 4, 5, 10])
+        with self.tracer.start_as_current_span("user_add_to_cart",
+                                            attributes={"user.id": user, "product.id": product, "quantity": quantity}):
+            logging.info(f"User {user} adding {quantity} of product {product} to cart")
+            self.client.get("/api/products/" + product)
+            cart_item = {
+                "item": {
+                    "productId": product,
+                    "quantity": quantity,
+                },
+                "userId": user,
+            }
+            self.client.post("/api/cart", json=cart_item)
 
     @task(1)
     def checkout(self):
         # checkout call with an item added to cart
         user = str(uuid.uuid1())
-        self.add_to_cart(user=user)
-        checkout_person = random.choice(people)
-        checkout_person["userId"] = user
-        self.client.post("/api/checkout", json=checkout_person)
+        with self.tracer.start_as_current_span("user_checkout_single", attributes={"user.id": user}):
+            logging.info(f"User {user} performing single item checkout")
+            self.add_to_cart(user=user)
+            checkout_person = random.choice(people)
+            checkout_person["userId"] = user
+            self.client.post("/api/checkout", json=checkout_person)
+            logging.info(f"Checkout completed for user {user}")
```

Comment:
All the new spans already increased a lot the amount of data leaving the load-gen service, let's keep just one log entry per call.

Reply:
Sure, just kept the final log statement for each entry.  [db6f18a](https://github.com/open-telemetry/opentelemetry-demo/pull/2265/commits/db6f18a483e5935a30209a39dde8f9c8e9211fae)